src/python/pants/backend/python/rules/pex.py

```diff
@@ -9,37 +9,45 @@
 from pants.backend.python.subsystems.python_native_code import PexBuildEnvironment
 from pants.backend.python.subsystems.python_setup import PythonSetup
 from pants.backend.python.subsystems.subprocess_environment import SubprocessEncodingEnvironment
-from pants.engine.fs import Digest
+from pants.engine.fs import (
+  EMPTY_DIRECTORY_DIGEST,
+  Digest,
+  DirectoriesToMerge,
+  DirectoryWithPrefixToAdd,
+)
 from pants.engine.isolated_process import ExecuteProcessResult, MultiPlatformExecuteProcessRequest
 from pants.engine.platform import Platform, PlatformConstraint
 from pants.engine.rules import optionable_rule, rule
 from pants.engine.selectors import Get
 
 
 @dataclass(frozen=True)
-class RequirementsPexRequest:
+class CreatePex:
+  """Represents a generic request to create a PEX from its inputs."""
   output_filename: str
-  requirements: Tuple[str, ...]
-  interpreter_constraints: Tuple[str, ...]
-  entry_point: Optional[str]
+  requirements: Tuple[str] = ()
+  interpreter_constraints: Tuple[str] = ()
```

Comment:
This should be `Tuple[str, ...]` to indicate arbitrary length. Otherwise this is a 1-tuple.